Qft

Lachnitt opened the pull request:
The quantum Fourier transform

Commit:
Added first ideas

Commit:
fixed some indices

Commit:
added intermediate proofs, improved formalization

Commit:
Proceeded with formalisation

Commit:
added intermediate proofs

Commit:
Tidied up formalisation

Commit:
Proceed adaption to new formalisation

Commit:
Added swap proofs

Commit:
unit_vec decomposition

Commit:
Proceed with basic formalisation

Commit:
Added proofs about swapping and unit_vec

Commit:
Save unit_vec_proof attempts before deleting them

Commit:
Proceed with unit_vec proof finished swap proofs

Commit:
Changed qr and adapted everything

Commit:
Proceed with proof of application of all R gates

Commit:
Changed indices

Commit:
Save before making major changes

Commit:
Finished proof about application of R_k

Commit:
Finished decomposition proofs, proceed with application of all Rks

Commit:
Finished proof application of all Rks

Commit:
Proceed to application of Rk on each qubit

Commit:
Added proof for tensor representation of quantum fourier transformation

Commit:
Merge branch 'master' of https://github.com/AnthonyBordg/Isabelle_marries_Dirac into QFT

Commit:
fixed index mistake

Commit:
Added gate proof CR

Commit:
proceed unit_vec to pow_tensor_list

Commit:
finished j_as_unit proof

Commit:
Tidied up first section

Commit:
Tryout proof ideas for second representation of QFT

Commit:
Started proof of second product representation

Commit:
Proceeded proof of second product representation

Commit:
Finished proof second product representation

Commit:
Tidied up proof of second product representation

Commit:
Proof equivalence to product representation wth case distinction

Commit:
Proof idea product representation to matrix

Commit:
Finished proof outline for product representation to matrix rep

Commit:
Finished proof of equivalence to product representation wth case distinction

Commit:
Started proof that product representation is equal to matrix definition

Commit:
added gate proofs

Commit:
Added gate proofs

Commit:
Tidied up

Commit:
Fixed remaining sorrys, cleaned up

Commit:
Fixed all remaining sorrys

Commit:
Fixed all remaining sorrys

Commit:
Tidied up

Commit:
Added authorship section

AnthonyBordg commented:
@Lachnitt 
Dear Hanna,

can you please update your PR ? 
Indeed, in the meantime a few names have been changed in my master branch, for instance `hermite_cnj` is now called `dagger` and so on.

Commit:
Merge branch 'master' of https://github.com/AnthonyBordg/Isabelle_marries_Dirac into QFT

Commit:
Updated names according to changes in Quantum.thy

Lachnitt commented:
Dear Anthony, 

I updated the PR. It should run now without any problems. However, I still need to adapt the lemma names used such that they meet the new naming conventions. E.g. there is a lemma named SWAP_front_hermite_cnj_dim.

Commit:
Renamed lemmas

Lachnitt commented:
@AnthonyBordg 
Renaming is finished.

AnthonyBordg committed:
Add files via upload

Commit:
Merge branch 'QFT' into qFt

Commit:
Merge pull request #22 from AnthonyBordg/qFt

Start reviewing qFt

AnthonyBordg commented:
@MH2520 If you want to review Hanna's PR with me, your help would be welcome.

AnthonyBordg committed:
Update

Commit:
Merge pull request #23 from AnthonyBordg/qFt

Update

Commit:
Adapted merge

Commit:
Changed controlled_phase_shift im lemma names to CR

Commit:
Changes lemma names

AnthonyBordg committed:
Update

- Rename m with n when m denotes the total number of qubits. But I haven't made the changes across the whole file, since it's very tedious.
- Rename k with l when k denotes a length.
- Rename zero_neq_one_mat with ket_zero_neq_ket_one.
- Next time, I will restart the reviewing process starting from subsection "The Controlled Phase Gate CR".

Commit:
Merge branch 'QFT' into qFt

Commit:
Merge pull request #24 from AnthonyBordg/qFt

Update

Commit:
proceeded with replacement of m by n

Commit:
Changed m into n

Commit:
Proceeded with replacement of m with n

Commit:
Finished renaming of n's to m's
